### ISSUE TITLE
Support for Initial Vector

### ISSUE THREAD

**Issue by kmusham** [johnpryan/tripledes-dart#7]
How do I pass the key and initialization vector to encrypt my string?

Please look at one example from Microsoft, I would like to mimic the same using your flutter APIs.

https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.tripledescryptoserviceprovider?view=netframework-4.7.2

Thanks

**Comment by kmusham** [comment]
is anyone maintaining these libraries? if so, please respond to the issue reported. Thanks.

**Comment by johnpryan** [comment]
Sorry I've been on vacation. Can you be more specific? Have you taken a look at the test/ directory for help?

**Comment by kmusham** [comment]
Yes, I did, but I did not see any API to pass initial vector along with a
key.

**Comment by johnpryan** [comment]
No, IVs are not currently supported. This package is based closely off of CryptoJS, so you could try to look at how CryptoJS implements.it https://gist.github.com/ufologist/5581486#file-cryptojs-des-html-L59

**Comment by lawDavid** [comment]
> How do I pass the key and initialization vector to encrypt my string?
> 
> Please look at one example from Microsoft, I would like to mimic the same using your flutter APIs.
> 
> https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.tripledescryptoserviceprovider?view=netframework-4.7.2
> 
> Thanks

I have same problem, have you found the way to transfer 3des with iv now?

**Comment by premtemp1** [comment]
I have the same issue.. Please let me know if you found  solution

**Comment by ghost** [comment]
Solution anyone???

**Comment by marianoarga** [comment]
Same problem here
